pkg/controller/sentryoperator/sentryoperator_controller.go

```diff
@@ -101,64 +113,42 @@ func (r *ReconcileSentryOperator) Reconcile(request reconcile.Request) (reconcil
 		return reconcile.Result{}, err
 	}
 
-	// Check if the deployment already exists, if not create a new one
-	found := &appsv1.Deployment{}
-	err = r.client.Get(context.TODO(), types.NamespacedName{Name: sentry.Name, Namespace: sentry.Namespace}, found)
-	if err != nil && errors.IsNotFound(err) {
-		// Define a new deployment
-		dep := r.newDeployment(sentry)
-		reqLogger.Info("Creating a new Deployment.", "Deployment.Namespace", dep.Namespace, "Deployment.Name", dep.Name)
-		err = r.client.Create(context.TODO(), dep)
-		if err != nil {
-			reqLogger.Error(err, "Failed to create new Deployment.", "Deployment.Namespace", dep.Namespace, "Deployment.Name", dep.Name)
+	requeue := false
+	for _, depName := range allDeployments {
+		// Check if the deployment already exists, if not create a new one
+		found := &appsv1.Deployment{}
+		err = r.client.Get(context.TODO(), types.NamespacedName{Name: depName, Namespace: sentry.Namespace}, found)
+		if err != nil && errors.IsNotFound(err) {
+			requeue = true
+			// Define a new deployment
+			var dep *appsv1.Deployment
+			switch {
+			case depName == SENTRY_WEB_UI:
+				dep = r.deploymentForSentryWebUI(sentry)
+			case depName == SENTRY_WORKER:
+				dep = r.deploymentForSentryWorker(sentry)
+			case depName == SENTRY_CRON:
+				dep = r.deploymentForSentryCron(sentry)
+			}
+			reqLogger.Info("Creating a new Deployment.", "Deployment.Namespace", dep.Namespace, "Deployment.Name", dep.Name)
+			err = r.client.Create(context.TODO(), dep)
```

Comment:
This seems to be creating the deployments no mater what. Are we properly checking for the existance of _each_ deployment?

Reply:
I See, thanks